sdk/sdk-api/src/main/java/mil/emp3/api/Point.java

```diff
@@ -30,39 +30,70 @@ public Point() {
      * provide the GeoIconStyle to correctly position the icon.
      * @param sURL A valid URL
      */
-    public Point(String sURL) {
+    public Point(final String sURL) {
         super(new GeoPoint(), FeatureTypeEnum.GEO_POINT);
-        this.setIconURI(sURL);
+        if(android.webkit.URLUtil.isValidUrl(sURL)) {
+            this.setIconURI(sURL);
+        } else {
+            throw new IllegalArgumentException("Invalid url input, " + sURL + " is not a valid url");
+        }
     }
 
     /**
      * This constructor creates a point feature positioned at the coordinates provided.
      * @param dLat
      * @param dLong
      */
-    public Point(double dLat, double dLong) {
+    public Point(final double dLat, final double dLong) {
         super(new GeoPoint(), FeatureTypeEnum.GEO_POINT);
-        
-        IGeoPosition oPos = new GeoPosition();
-        oPos.setLatitude(dLat);
-        oPos.setLongitude(dLong);
+        final IGeoPosition oPos = new GeoPosition();
+        if (isValidLat(dLat)) {
+            oPos.setLatitude(dLat);
+        }
+        if(isValidLong(dLong)) {
+            oPos.setLongitude(dLong);
+        }
         this.setPosition(oPos);
     }
 
     /**
      * This constructor creates a point feature from a geo point object.
      * @param oRenderable
      */
-    public Point(IGeoPoint oRenderable) {
+    public Point(final IGeoPoint oRenderable) {
         super(oRenderable, FeatureTypeEnum.GEO_POINT);
     }
 
+    private boolean isValidLat(final Double dLat) {
```

Comment:
this method should be called "validateLatitude" and not return anything because it can never return false.  The exceptions are thrown instead. this also applies to your other validation functions

Reply:
also please add comments to all methods